### ISSUE TITLE
[jss-default-unit] Allow value updating before appending unit

### ISSUE THREAD

**Issue by jkjustjoshing** [cssinjs/jss#1291]
# Problem
I'm trying to switch from `px` units to `rem` units. I've set `html { font-size: 62.5%; }` as described in many blog posts, including [this one](https://engageinteractive.co.uk/blog/em-vs-rem-vs-px). In this situation, `16px` is equivalent to `1.6rem`. I want to be able to change the options passed to `defaultUnit` without changing any values in my code.

# Solution 
To do this, I want to switch the default unit for all properties that have default `px` values, but also divide the value by 10. The built-in options for the plugin allows the setting to be a string to append to the value, or a function that takes the value and returns the string. However, when you pass options to the plugin as a consumer of the plugin, it doesn't let you pass a function. 

I propose that we allow the plugin consumer to specify transform functions instead of just strings, so I can do something like this:
```
defaultUnit({
    width: (val) => (val / 10) + 'rem'
})
```

# Willing to implement?
I could probably throw a PR together.

**Comment by kof** [comment]
Would be using font-size on root with some value that translates to int values with rem an option?

**Comment by jkjustjoshing** [comment]
If i understand you correctly, you're asking if I can set `html { font-size: 6.25%; }` to make `1px` equivalent to `1rem`?

That's not a terrible suggestion, but it's *extremely* uncommon to do that compared to setting it to `62.5%`, i believe due to issues with font sizes less than 10% in some browsers? I'm not an expert, just doing some Googling around,  and it seems like the best solution for me is to divide bare units by 10 in my JSS. 

Is there a reason not to implement this? Should be as simple as doing
```
const unit = option [prop] || units[prop];
```
and then using `unit` instead of `units[prop]` and removing the specific
```
options[prop]
```
conditional. Actually I'll just throw the PR together now for the heck of it.

**Comment by kof** [comment]
released in 10.1.0
